### PR TITLE
Recyclerview

### PR DESCRIPTION
**Pull request by RahulChakravarthy**

### CODE REVIEW

File: app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt [hunk 1]
```diff
@@ -7,24 +7,41 @@ import android.view.ViewGroup
 import androidx.recyclerview.widget.RecyclerView
 import com.hacks.radish.R
 import com.hacks.radish.repo.dataobject.FeedDO
+import com.hacks.radish.views.FeedCardView
+import kotlinx.android.synthetic.main.adapter_feed_view.view.*
+import java.util.*
 
-class FeedAdapter(val list : ArrayList<FeedDO>, val context : Context) : RecyclerView.Adapter<FeedAdapter.ViewHolder>() {
+class FeedAdapter(var list : ArrayList<FeedDO>, val context : Context) : RecyclerView.Adapter<FeedAdapter.ViewHolder>() {
```
Comment (PatzHum) [inline app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt]: Don't make this a `var` if we're listening for updates. Consider doing it like this
```kt
FeedAdapter(initialList : ArrayList<FeedDO>, val context : Context) : RecyclerView.Adapter<FeedAdapter.ViewHolder>() {
    var list: ArrayList<FeedDO> = initialList
        set(value) {
            field = value
            notifyDataSetChanged()
        }
```

File: app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt [hunk 2]
```diff
@@ -7,24 +7,41 @@ import android.view.ViewGroup
...
 
-    class ViewHolder(private val view : View) : RecyclerView.ViewHolder(view) {
+    class ViewHolder(private val view : View) : RecyclerView.ViewHolder(view)
 
+    fun updateFeedList(new : ArrayList<FeedDO>) {
```
Comment (PatzHum) [inline app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt]: If you take the above commend you don't need this

File: app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt (hunk 1 above)
Comment (PatzHum) [reply app/src/main/java/com/hacks/radish/adapters/FeedAdapter.kt]: The problem with the previous way was that that you publicly expose `list` which causes a mismatch with the data in the adapter vs the data shown.

**Review by PatzHum — CHANGES_REQUESTED** [review]

File: app/src/main/res/xml/network_security_config.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <domain-config cleartextTrafficPermitted="true">
```
Comment (PatzHum) [inline app/src/main/res/xml/network_security_config.xml]: Will we still need this when HTTPs is added on backend?

File: app/src/main/res/layout/adapter_feed_view.xml [hunk 4]
```diff
@@ -1,24 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
                                                    xmlns:app="http://schemas.android.com/apk/res-auto"
-                                                   android:orientation="vertical"
```
Comment (PatzHum) [inline app/src/main/res/layout/adapter_feed_view.xml]: Are we fixing orientation in manifest for the whole app? I think we should, even Uber doesn't support landscape

File: app/src/main/java/com/hacks/radish/viewmodels/MainActivityViewModel.kt [hunk 5]
```diff
@@ -34,73 +32,40 @@ class MainActivityViewModel @Inject constructor(private val context : Context,
...
     init {
         dashboardFragments.add(PostFragment())
         dashboardFragments.add(FeedFragment())
     }
 
+    fun fetchNewFeed(size : Int) {
+//        apiManager.getFeed(size) {
+//            feedListLiveData.value = it
+//        }
+        feedListLiveData.value = Array(5) {FeedDO.generateTestFeedDO()}.toCollection(ArrayList())
```
Comment (PatzHum) [inline app/src/main/java/com/hacks/radish/viewmodels/MainActivityViewModel.kt:59]: 5?

Comment (RahulChakravarthy) [reply app/src/main/java/com/hacks/radish/viewmodels/MainActivityViewModel.kt:59]: just a test value until backend fixes their stuff

**Review by PatzHum — APPROVED** [review]

**Pull request merged**